### ISSUE TITLE
TLS/SSL support not available; install glib-networking

### ISSUE THREAD

**Issue by joselluis7** [eneshecan/whatsapp-for-linux#54]
I'm getting the error message **TLS/SSL support not available; install glib-networking** After install whatsapp for linux in my fedora 31, even though I have already installed glib-networking-2.62.4-1

**Comment by eneshecan** [comment]
Hi @joselluis7. Thank you for the feedback. Which package of the app are you using?

**Comment by joselluis7** [comment]
snap

**Comment by joselluis7** [comment]
snapd is the daemon

**Comment by eneshecan** [comment]
Needs to be looked into. Thanks.

**Comment by n1snt** [comment]
#90 Check if this fixes it @joselluis7

**Comment by eneshecan** [comment]
@n1snt Well, I haven't experienced the issue with the snap we built. Needs to be rebuilt, installed and checked if it persists. However, if glib-networking doesn't come with snap build as default, this would be the right solution.

**Comment by eneshecan** [comment]
Here it mentions the same problem in the first reply; https://forum.snapcraft.io/t/webkit2gtk-error-failed-to-load-module-canberra-gtk-module/21624/2. It seems that it doesn't fix the problem. Ahh, I hate snaps :/

**Comment by eneshecan** [comment]
Let's give it a try. I will merge it.

**Comment by SCOTT-HAMILTON** [comment]
Hello, I get this same issue with whatsapp-for-linux package 1.1.7 on NixOS : 

```shell_session
$ nix-shell -I 'nixpkgs=.' -p whatsapp-for-linux glib-networking # Creates an environment with latest whatsapp-for-linux and glib-networking
$ whatsapp-for-linux
```
Output : 
```
Settings: Key file does not have key “allow_permissions” in group “Network”
Settings: Key file does not have key “close_to_tray” in group “General”
Settings: Key file does not have key “start_in_tray” in group “General”
Settings: Key file does not have key “start_in_tray” in group “General”
```

Screen : 

<details>

![image](https://user-images.githubusercontent.com/24496705/121774512-029e1180-cb83-11eb-9a8f-1114a97e8e85.png)


</details>

**Comment by michaelerule** [comment]
I'm seeing the same error again (May 2025) on `Ubuntu 11.04.1 LTS`. 

This is from WasIstLos built from source 4–12 months ago (sadly I deleted the build directory and can't find a way to get the current binary to report version). 

The app was previously working, so I presume that either changes to the system packages (hard to pinpoint when) or the protocol WaIstLos is talking to have changed? 

I ran `sudo apt-get install --reinstall glib-networking` (`2.72.0-1 amd64`), but this did not fix it. 

For reference, here are the package numbers for the dependencies:
    
    cmake:amd64/jammy 3.22.1-1ubuntu1
    intltool:all/jammy 0.51.0-6
    libgtkmm-3.0-dev:amd64/jammy 3.24.5-1build1
    libwebkit2gtk-4.1-dev:amd64/jammy-security 2.48.1-0ubuntu0.22.04.1
    libayatana-appindicator-dev:amd64/jammy 0.5.90-7ubuntu2
    libcanberra-dev:amd64/jammy 0.30-10ubuntu1
    libhunspell-dev:amd64/jammy 1.7.0-4build1
    glib-networking:amd64/jammy 2.72.0-1
    glib-networking:i386/jammy 2.72.0-1
    glib-networking-common:all/jammy 2.72.0-1
    glib-networking-services:amd64/jammy 2.72.0-1
    glib-networking-tests:amd64/jammy 2.72.0-1

Further steps: 

 - I uninstalled, re-cloned the repository (on 2025.05.02, commit short hash `a2d3732`).
 - I re-built per the `README.md` instructions using the command `cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/usr ../.. && make -j4`. 
 - Possible discrepancy: There was one warning about `webkit_web_view_run_javascript` being deprecated and there was no rule to `make update-translation`. 
 - Possible discrepancy: The `wasistlos` binary ended up `./build/debug/src/` rather than `./build/debug/` as implied by the ability to run `./wasistlos` without changing directory after building in `README.md`.

The newly build binary loads with a white page with the text `"TLS support is not available"`, so it seems the behavior is unchanged from before. 

This appears to be a generic error, so I'm not sure this is the correct thread. 

 - [This answer](https://stackoverflow.com/a/77266263/900749) is perhaps related, although I don't quite understand it yet. 
 - [This Github thread](https://github.com/GSConnect/gnome-shell-extension-gsconnect/issues/1951) suggests that seeing this error with `glib-networking` installed might indicate that the command `openssl ciphers` is missing. However, `openssl ciphers` exists and prints a number of ciphers on my system. 
 - [@dmedinag's](https://github.com/dmedinag) note (above) is for Nix, and I'm not sure what the analogous form for Ubuntu would be. The `networking` service is up. All other apps can access the internet. `sudo systemctl restart NetworkManager` restarts the network but does not resolve the TLS error. 


Any ideas?
